Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -802,6 +802,18 @@ void BlockFormattingContext::place_block_level_element_in_normal_flow_vertically
     box_state.set_content_offset(CSSPixelPoint { box_state.offset.x(), y });
 }
 
+// Returns whether the given box has the given ancestor on the path to root, ignoring the anonymous blocks.
+static bool box_has_ancestor(Box const* box, Box const* ancestor, Box const* root)
```

Comment:
Feels like `box_has_ancestor` hides the fact that this only considers non-anonymous containing blocks.
Perhaps something more explicit, like `box_has_ancestor_in_non_anonymous_containing_block_chain()`?
Also, if a parameter can't be null, please use a reference type, i.e `Box const&` instead of `Box const*` :)

Reply:
Sounds good, done.